frame/system/benchmarking/Cargo.toml

```diff
@@ -15,9 +15,9 @@ targets = ["x86_64-unknown-linux-gnu"]
 [dependencies]
 codec = { package = "parity-scale-codec", version = "3.6.1", default-features = false }
 scale-info = { version = "2.5.0", default-features = false, features = ["derive"] }
-frame-benchmarking = { version = "4.0.0-dev", default-features = false, path = "../../benchmarking" }
-frame-support = { version = "4.0.0-dev", default-features = false, path = "../../support" }
-frame-system = { version = "4.0.0-dev", default-features = false, path = "../../system" }
+frame-benchmarking = { version = "4.0.0-dev", default-features = false, path = "../../benchmarking", features = ["runtime-benchmarks"] }
+frame-support = { version = "4.0.0-dev", default-features = false, path = "../../support", features = ["runtime-benchmarks"] }
+frame-system = { version = "4.0.0-dev", default-features = false, path = "../../system", features = ["runtime-benchmarks"] }
```

Comment:
>A required feature flag doesn't make sense.

Rust feature flags are additive, and this crate only does benchmarking. So without the flag the crate just does nothing (which is fine).

We cannot ever enable `runtime-benchmarks` or `try-runtime` by default. It causes all kinds of issues by being propagated downwards. There is a script in the CI to prevent this as well.

Reply:
I don't 100% get what you mean. Crates can be optional. You shouldn't include/enable this if it is not meant for runtime benchmarks.

Reply:
That means we are shifting the problem from not accidentally enabling the feature to not accidentally enabling this crate?  
I think just never enabling `runtime-benchmarks` and `try-runtime` by default is a pretty simple rule, that does not need any exceptions. But yea, it should also be possible to optionally enable the dependency.

Reply:
Include an empty crate doesn't make a lot of sense to me. Why do you ever want to do that?

> I think just never enabling runtime-benchmarks and try-runtime by default is a pretty simple rule, that does not need any exceptions. 

I don't agree with this. Pallets specifically built for runtime benchmarks (i.e. this one) can have runtime-benchmarks enabled by default.

On another note, I don't like implicit rules. If it is not sufficiently discussed, and documented, they don't exist to me.